# Assignment 19.2

Using the following data, perform a oneway analysis of variance using α=.05. Write up the results in APA format.

[Group1: 51, 45, 33, 45, 67]

[Group2: 23, 43, 23, 43, 45]

[Group3: 56, 76, 74, 87, 56]

# One Way Aanalysis

A one way analysis is used to compare two means from two independent (unrelated) groups using the F-distribution. 
The null hypothesis for the test is that the two means are equal. Therefore, a significant result means that the two means 
are unequal.

The ANOVA test has important assumptions that must be satisfied in order for the associated p-value to be valid.

1. The samples shall be independent.
2. Each sample shall be from a normally distributed population.
3. The population standard deviations of the groups shall be equal. This property is known as homoscedasticity.


In [1]:
import pandas as pd
import scipy.stats as stat

In [22]:
data = pd.DataFrame({'Group1':[51,45,33,45,67], 'Group2':[23,43,23,43,45], 'Group3':[56,76,74,87,56]})
data

Group1  Group2  Group3
0      51      23      56
1      45      43      76
2      33      23      74
3      45      43      87
4      67      45      56

In [4]:
data.describe()

Group1     Group2     Group3
count   5.000000   5.000000   5.000000
mean   48.200000  35.400000  69.800000
std    12.377399  11.349009  13.535139
min    33.000000  23.000000  56.000000
25%    45.000000  23.000000  56.000000
50%    45.000000  43.000000  74.000000
75%    51.000000  43.000000  76.000000
max    67.000000  45.000000  87.000000

We can see that standard deviations are close to each other. We can perform ANOVA testing

# Limitations of the One Way analysis

A one way analysis will tell us that at least two groups were different from each other. But it won’t tell us what groups
were different. If our test returns a significant f-statistic, we may need to run an ad hoc test (like the Least Significant 
Difference test) to tell us exactly which groups had a difference in means.

# Solution:

1. Defining Null Hypothesis , Alternate Hypothesis and Alpha  
H0 (Null Hypothsis)      :  Mean of Group1, Group2 and Group3 are equal

    H1 (Alternate Hypothesis):  Mean of Group1, Group2 and Group3 are not equal 
    
Rejection Criteria:
If alpha(0.05) < F

It means that if F values as in table and as calculated are not equal, we will reject the Null Hypothesis

In [7]:
mean = (data.sum().sum()) / 15
mean

51.13333333333333

We can see that in our calculation mean of 3 Groups is 51.1333, let us check F-statistic value too

In [8]:
data.var()

Group1    153.2
Group2    128.8
Group3    183.2
dtype: float64

In [9]:
ms_error = (153.2+128.8+183.2) / 3       # mean sum of squares
ms_error

155.06666666666666

In [10]:
n = 15                ## n is total of all the data sets combined
k = 3                 # k is the numbers of groups
deg_freedom = n-k     ## is the degrees between groups
deg_freedom

12

In [11]:
ss_error = ms_error * deg_freedom     ## sum of squares error
ss_error

1860.8

In [12]:
data.sum().sum()

767

In [13]:
sq_dev = (data.mean()-mean)**2     ## square deviations
sq_dev

Group1      8.604444
Group2    247.537778
Group3    348.444444
dtype: float64

In [15]:
ss_means = (8.604+247.537+348.444)     ## sum of square deviation
ss_means

604.585

In [16]:
var_means = ss_means/(3-1)
var_means

302.2925

In [18]:
ms_between= var_means * 5        # Between group variance is the between group variation divided by its degrees of freedom
ms_between

1511.4625

In [19]:
group_freedom = k-1                         # degree of freedom of groups 

ss_group = ms_between * group_freedom  

ss_group                                    # calculating the remaining between(or group) terms of the ANOVA table

3022.925

In [20]:
F = ms_between / ms_error            # Test Statistic
F

9.747178632846088

# Alternate Method

In [24]:
Group1 = [51, 45, 33, 45, 67]
Group2 = [23, 43, 23, 43, 45]
Group3 = [56, 76, 74, 87, 56]

In [25]:
stat.f_oneway(Group1,Group2,Group3)

F_onewayResult(statistic=9.747205503009463, pvalue=0.0030597541434430556)

Since pvalue < 0.05. The differences between the means are statistically significant. Therefore, we reject the null 
hypothesis that mean of Group1, Group2 and Group3 are equal.

In [26]:
chi_square = ss_group / (ss_error + ss_group)
chi_square

0.6189793651362433

# Relationship between p-value and chi-square.

The "Chi-square" calculates how much the 'observed' data differs from what we would 'expect' if 
there were no real difference at all (i.e. Alternate Hypothesis no relationship).  If groups are related, ch-sq will be large.
If not related, the chi-square will be very small.  

The final result is written with a measure of how likely the null hypothesis was true, and by convention, if it was 5% or 
less likely to be correct, we are allowed to claim the result as "statistically significant".  THIS is expressed as
<0.05 or <0.01  etc., depending on the precise numbers.   

This is the "rejection level" or "probability statement" or "alpha level", or "P-value".  
So we can see that the ch-sq is the statistical measurement, while the P value is the level of probability that the result 
was due to chance alone.  As the chi-sq statistic becomes larger, the P value becomes smaller.


# APA Writeup:
F(2,12) = 9.75  , p < 0.05 , chi_square=0.6189